### PR TITLE
[UC19#134] Implemented Exchanges Page to allow users to view an interface containing their own exchange proposals, both sent and received

### LINKED ISSUE #134: Creare ProposalListWidget della lista delle proposte e impostare UI pagina delle proposte [UC19]
_**Dod**_
- [x]  La pull request ha un titolo e descrizione adeguata
- [x]  La pull request è stata collegata con la issue del task
- [x]  I requisiti funzionali del task collegato sono soddisfatti
- [x]  Il codice si compila senza errori
- [x]  Gli Unit Tests sono stati scritti e passati 
- [x]  Gli Unit Tests esistenti sono stati passati
- [x]  Il Code Review è stato fatto
- [x]  La documentazione è stata aggiornata
- [x]  La pull request è stata accettata e fatto merge sul branch development.

 **Effort effettivo:**

### PR DESCRIPTION
**Pull request by alessiacrimaldi**
This PR implements #134 creating an interface for user visualization of sent and received exchange proposals. A widget named `ProposalListWidget` has been created for displaying the two lists of proposals 'From you' and 'To you', passing them the appropriate parameters (the title, the other user type and the list of proposals contained). The rows of the two tables show a tooltip when the mouse is over and are clickable. For now, the onClickProposalRow is a dummy method which shows an informative alert and takes the id of the proposal as parameter: this will be useful in task #136 when the user will be navigated to the Proposal Page.

### CODE REVIEW

**Review by alessioarcara — APPROVED** [review]
Overall, the code looks good and I have no major concerns. However, I noticed that the `userList` and `ProposalList` widgets are very similar and could potentially be combined into a single widget. This would simplify the code and make it more efficient. This issue is not critical, but it is something to consider for future improvement. Great job!

File: src/main/java/com/aadm/cardexchange/client/views/ExchangesViewImpl.java [hunk 1]
```diff
@@ -1,19 +1,60 @@
...
+    public void setToYouProposalList(List<Proposal> proposals) {
+        proposals.forEach(proposal ->
+                toYouProposalList.addRow(proposal.getId(), "08-02-2023", proposal.getSenderUserEmail(), this)
+        );
+    }
+
+    @Override
+    public void onClickProposalRow(int selectedProposalId) {
+        // There will be the code for opening the Proposal Page
+        Window.alert("This is the " + selectedProposalId + " proposal");
```
Comment (Matte0Sacc0) [inline src/main/java/com/aadm/cardexchange/client/views/ExchangesViewImpl.java:46]: I leave a comment just to keep it in mind...
This part will be updated soon with the real navigation to Proposal page

**Review by Matte0Sacc0 — APPROVED** [review]
Great Job

We need to check if there is already a task for the navigation to Proposal page and if not create it to update the little snippet in the future without forgetting

File: src/main/java/com/aadm/cardexchange/client/views/ExchangesViewImpl.java (hunk 1 above)
Comment (alessiacrimaldi) [reply src/main/java/com/aadm/cardexchange/client/views/ExchangesViewImpl.java:46]: Yes of course, I mentioned the task in the description

**Pull request merged**